modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/capabilities/TemporaryFileEntriesCapabilityImpl.java

```diff
@@ -344,26 +345,30 @@ private void _runWithoutSystemEvents(
 
 		@Override
 		public void execute(FileEntry fileEntry) throws PortalException {
-			Folder folder = fileEntry.getFolder();
+			DLAppHelperLocalServiceUtil.deleteFileEntry(fileEntry);
 
 			_documentRepository.deleteFileEntry(fileEntry.getFileEntryId());
 
+			Folder folder = fileEntry.getFolder();
+
 			Folder mountFolder = _documentRepository.getFolder(
 				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID);
 
 			while (folder.getFolderId() != mountFolder.getFolderId()) {
+				long folderId = folder.getFolderId();
+
 				int count = _documentRepository.getFileEntriesCount(
-					folder.getFolderId(), WorkflowConstants.STATUS_ANY);
+					folderId, WorkflowConstants.STATUS_ANY);
 
 				if (count != 0) {
 					break;
 				}
 
-				long folderId = folder.getFolderId();
-
-				folder = folder.getParentFolder();
+				DLAppHelperLocalServiceUtil.deleteFolder(folder);
 
 				_documentRepository.deleteFolder(folderId);
+
+				folder = folder.getParentFolder();
```

Comment:
Putting this after the delete call may cause problems. To be on the safe side, we shouldn't assume the getParentFolder() call doesn't depend on the folder existing. If we change the implementation this may break.

Did you change the order for some specific reason? Or is simply because updating the loop variable at the end is cleaner? If the reason is the latter, I'll change the order and send it to Brian.

Reply:
Thank for helping me with this @adolfopa !